### ISSUE TITLE
copy-frameworks Script Doesn't Seem to be working

### ISSUE THREAD

**Issue by b-lester** [Carthage/Carthage#1859]
* `carthage version`: 0.20.1
* `xcodebuild -version`: Xcode 8.2.1 Build version 8C1002
* Are you using `--no-build`? No
* Are you using `--no-use-binaries`?  No
* Are you using `--use-submodules`? No
* Are you using `--cache-builds`? No

**Cartfile**
```
# Networking framework used in file upload code
github "AFNetworking/AFNetworking" ~> 3.0

# Analytics tracking framework
github "adjust/ios_sdk"
```

**Carthage Output**
```
Not Applicable??
```

We're trying to submit our app to the app store and we ran into a problem which boiled down to the copy-frameworks script. It wasn't stripping out the unsupported architectures and signing as it was supposed to. 

We added the script to the build phases exactly as described on the main GitHub page but it simply wasn't working.

However, I did find [this post](https://github.com/Carthage/Carthage/issues/188#issuecomment-66267828) containing a script which I understood as the bash equivalent to the current swift-based copy-frameworks and after putting that (with minor modifications) into its own Run Script phase, everything started working properly and we were able to upload builds to TestFlight again. 

Long story short, the copy-frameworks script doesn't seem to work at all or we're missing something.

**Comment by mdiep** [comment]
That's weird because it's definitely working for other people. It's hard to say what could be going on here without access to a project that reproduces the issue.

**Comment by CiraciNicolo** [comment]
I think I'm facing the same issue. I can upload a build to iTC but then Apple says there are some problem with my dSYMs' build and therefore it's invalid. Running `carthage update` seems to fix the issue. I can only speculate that Apple changed something with the elaboration of dSYMs and they need 'a fresh copy' of them for each build.

**Comment by oscart** [comment]
I'm facing the same issue here.

**Comment by b-lester** [comment]
Here is the script we ended up going with, feel free to adapt it to your requirements (obviously you'll have to add/remove $SCRIPT_INPUT_FILE_X as needed).

We will revisit using the built in copy_frameworks script but for now this is working for us

`# Script to strip & sign frameworks built by Carthage

copy_framework ()
{
    local framework=$1
    local frameworks_folder="${CONFIGURATION_BUILD_DIR}/${FRAMEWORKS_FOLDER_PATH}"
    
    mkdir -p "${CONFIGURATION_BUILD_DIR}/${FRAMEWORKS_FOLDER_PATH}"
    rsync -av "$framework" "$frameworks_folder"
    
    local framework_path="$frameworks_folder/`basename $framework`"
    local file_path="$framework_path/`basename $framework .framework`"
    
    if [[ ! "$VALID_ARCHS" =~ "i386" ]];
    then
    lipo -remove i386 "$file_path" -output "$file_path"
    fi
    
    if [[ ! "$VALID_ARCHS" =~ "x86_64" ]];
    then
    lipo -remove x86_64 "$file_path" -output "$file_path"
    fi
    
    if [[ "$CODE_SIGNING_REQUIRED" -eq YES ]];
    then
    codesign --force --sign "$EXPANDED_CODE_SIGN_IDENTITY" --preserve-metadata=identifier,entitlements,resource-rules "$framework_path"
    fi
}

copy_framework $SCRIPT_INPUT_FILE_0
copy_framework $SCRIPT_INPUT_FILE_1

`

**Comment by Shunzi007** [comment]
I am use cmake to build an swift app.
I find it is hard to use command ` copy-frameworks` in CmakeList.txt.
Could you give me some suggestion about use this command in cmake?
